api/v1/openapi.yaml

```diff
@@ -946,6 +946,31 @@ definitions:
         description: |
           Whether to build an endpoint synchronously
         type: boolean
+      datapath-configuration:
+        "$ref": "#/definitions/EndpointDatapathConfiguration"
+  EndpointDatapathConfiguration:
+    description: Datapath configuration to be used for the endpoint
+    type: object
+    properties:
+      require-arp-passthrough:
+        description: Enable ARP passthrough mode
+        type: boolean
+      require-egress-prog:
+        description: >
+          Endpoint requires a host-facing egress program to be attached to
+          implement ingress policy and reverse NAT.
+        type: boolean
+      external-ipam:
+        description: >
+          Indicates that IPAM is done external to Cilium. This will prevent the
+          IP from being released and re-allocation of the IP address is skipped
+          on restore.
+        type: boolean
+      disable-routing:
```

Comment:
Do we want to have negatively-defined options in the API?

Reply:
Existing clients won't define this option, how else do you want to do it?

Reply:
@tgraf something like
```yaml
      enable-routing:
        description: >
          By enabling routing, all routing decisions are delegated to the
          cilium routing layer.
        type: boolean
        default: true
```

```golang
// RoutingEnabled returns false if...
func (e *Endpoint) RoutingEnabled() bool {
	return e.DatapathConfiguration.EnableRouting != nil && e.DatapathConfiguration.EnableRouting
}
```

Reply:
I've been trying to avoid that to not introduce `*bool` but I'm OK with it. I agree that it is cleaner from an API perspective.